### PR TITLE
display enrollment tax household details on enrollment summary page

### PR DESCRIPTION
**Pull request by saipraveen18**
# PR Checklist

Please check if your PR fulfills the following requirements:
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [X] Tests for the changes have been added (for bug fixes/features), and they use `let` helpers and `before` blocks.
- [ ] For all UI changes, there is Cucumber coverage.
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, the reasoning is documented in the PR and code.
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run them is documented in both the PR and the code.
- [ ] There are no inline styles added.
- [ ] There is no inline JavaScript added.
- [ ] There is no hard-coded text added/updated in helpers/views/JavaScript. New/updated translation strings do not include markup/styles unless there is supporting documentation.
- [ ] Code does not use `.html_safe`.
- [ ] All images added/updated have alt text.
- [ ] Does not bypass RuboCop rules in any way.

# PR Type
What kind of change does this PR introduce?:

- [ ] Bugfix
- [X] Feature (requires Feature flag)
- [ ] Data fix, Migration, or Report (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/188496814

# A brief description of the changes:

Current behavior: On the Plan summary page, if the Plan has APTC applied, it is becoming difficult for the business to identify which FAA application's APTC was actually used. 

New behavior: To easily identify which FAA application's APTC was used for the Plan, we are displaying Application details and Enrollment tax household details on the Plan summary page.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and indicate which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context

Screenshot 2024-11-12 at 9 30 14 AM

### CODE REVIEW

**Review by saikumar9 — DISMISSED** [review]
lgtm!

File: app/views/ui-components/v1/cards/_summary.html.erb [hunk 1]
```diff
@@ -49,17 +54,45 @@
...
+          </dl>
+
+          <h2 data-cuke="tax_household_enrollment_details"><%= l10n('tax_household_enrollment.details') %></h2>
+          <% @hbx_enrollment.tax_household_enrollments.each_with_index do |thh_enr, index| %>
+            <h3><%= "#{l10n('tax_household_enrollment')} #{index + 1}" %></h3>
+            <dl>
+              <dt><%= l10n('tax_household_enrollment.members') %></dt>
+              <dd><%= thh_enr.tax_household_enrollment_member_names %></dd>
+              <dt><%= l10n('tax_household_enrollment.health_plan_benchmark_value') %></dt>
+              <dd><%= thh_enr.household_health_benchmark_ehb_premium || l10n('not_available') %></dd>
```
Comment (raghuramg) [inline app/views/ui-components/v1/cards/_summary.html.erb:84]: you may want use `number_to_currency` helper here, for displaying amounts. same with `household_dental_benchmark_ehb_premium`

**Review by raghuramg — CHANGES_REQUESTED** [review]
requested changes

**Review by saikumar9 — DISMISSED** [review]
LGTM!

File: app/models/hbx_enrollment.rb [hunk 2]
```diff
@@ -2976,6 +2976,13 @@ def aptc_tax_household_enrollments
     end
   end
 
+  def tax_household_group
+    thh_enrs = tax_household_enrollments
+    return if thh_enrs.blank?
+
+    thh_enrs.first&.tax_household&.tax_household_group
```
Comment (raghuramg) [inline app/models/hbx_enrollment.rb:2983]: can we use most recent record...`thh_enrs.last` instead of `thh_enrs.first`. will that make any difference

**Review by saikumar9 — APPROVED** [review]

**Review by raghuramg — APPROVED** [review]
LGTM

**Pull request merged**